packages/compiler/src/core/source-loader.ts

```diff
@@ -164,6 +169,7 @@ export async function createSourceLoader(
     if (options?.getCachedScript) {
       const old = options.getCachedScript(file);
       if (old?.file === file && deepEquals(old.parseOptions, options.parseOptions)) {
+        mutate(old).importedBy = [];
```

Comment:
@timotheeguerin , I am not very comfortable modifying the old object here, but I can't find a better place to put the importedBy information. Any suggestion here?